### ISSUE TITLE
BiocStyle::html_document not working with rmarkdown 1.2

### ISSUE THREAD

**Issue by JokingHero** [Bioconductor/BiocStyle#19]
Hi! 

I currently have no idea what is wrong, but it seems its either rmarkdown or BiocStyle problem with the latest versions. When I used older versions it worked, also changing to BiocStyle::html_document2 works. I haven't found anything in the internet on this problem, checked on my windows with the same problem. 

My test.Rmd file contents:

```
---
title: "why me no good"
output: BiocStyle::html_document
vignette: >
  %\VignetteIndexEntry{test}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---
```

When trying to knit file I get:

 ```
 |.................................................................| 100%
   inline R code fragments

processing file: test.Rmd
output file: test.knit.md

Error in rmarkdown:::pandoc_html_highlight_args(highlight, template = "default",  : 
  unused arguments (self_contained, lib_dir, output_dir)
Calls: <Anonymous> -> <Anonymous> -> overlay -> <Anonymous>
Execution halted


> sessionInfo()
R version 3.3.1 (2016-06-21)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.1 LTS

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] backports_1.0.4 BiocStyle_2.2.0 magrittr_1.5    rprojroot_1.1   htmltools_0.3.5 tools_3.3.1     yaml_2.1.14    
 [8] Rcpp_0.12.8     stringi_1.1.2   rmarkdown_1.2   knitr_1.15.1    stringr_1.1.0   digest_0.6.10   evaluate_0.10  
```

**Comment by aoles** [comment]
Thank you @JokingHero for your report! This is because of some changes to **rmarkdown**'s internal API, as indicated by @yihui in https://github.com/Bioconductor/BiocStyle/issues/18. I will issue a patch ASAP.

In the meantime, you are welcome to have a look at `BiocStyle::html_document2` which will supersede `BiocStyle::html_document` in in the course of the next months.

**Comment by aoles** [comment]
Fixed in [`307ad19`](https://github.com/Bioconductor-mirror/BiocStyle/commit/307ad19).
